File: _code-samples/calculate-reserves/java/CalculateReserves.java

// Set up client ----------------------

import okhttp3.HttpUrl;
import org.xrpl.xrpl4j.client.JsonRpcClientErrorException;
import org.xrpl.xrpl4j.client.XrplClient;
import org.xrpl.xrpl4j.model.client.accounts.AccountInfoRequestParams;
import org.xrpl.xrpl4j.model.client.common.LedgerSpecifier;
import org.xrpl.xrpl4j.model.transactions.Address;
import java.math.BigDecimal;

public class CalculateReserves {
    public static void main(String[] args) throws JsonRpcClientErrorException {
        HttpUrl rippledUrl = HttpUrl.get("https://s.devnet.rippletest.net:51234/");
        XrplClient xrplClient = new XrplClient(rippledUrl);

// Look up reserve values ----------------------

        var serverInfo = xrplClient.serverInformation();
        var validatedLedger = serverInfo.info().validatedLedger().get();

        BigDecimal baseReserve = validatedLedger.reserveBaseXrp().toXrp();
        BigDecimal reserveInc  = validatedLedger.reserveIncXrp().toXrp();

        System.out.println("Base reserve: " + baseReserve + " XRP");
        System.out.println("Incremental reserve: " + reserveInc + " XRP");

// Look up owner count ----------------------

        Address address = Address.of("rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh");
        var accountInfo = xrplClient.accountInfo(
            AccountInfoRequestParams.builder()
                .account(address)
                .ledgerSpecifier(LedgerSpecifier.VALIDATED)
                .build()
        );

        long ownerCount = accountInfo.accountData().ownerCount().longValue();

// Calculate total reserve ----------------------

        BigDecimal totalReserve = baseReserve.add(reserveInc.multiply(BigDecimal.valueOf(ownerCount)));

        System.out.println("Owner count: " + ownerCount);
        System.out.println("Total reserve: " + totalReserve + " XRP");
    }
}

File: _code-samples/calculate-reserves/go/calculate_reserves.go

// Set up client ----------------------
package main

import (
	"fmt"
	"github.com/Peersyst/xrpl-go/xrpl/queries/account"
	"github.com/Peersyst/xrpl-go/xrpl/queries/server"
	"github.com/Peersyst/xrpl-go/xrpl/transaction/types"
	"github.com/Peersyst/xrpl-go/xrpl/websocket"
)

func main() {
	client := websocket.NewClient(
		websocket.NewClientConfig().
			WithHost("wss://s.devnet.rippletest.net:51233"),
	)
	defer client.Disconnect()

	if err := client.Connect(); err != nil {
		panic(err)
	}

// Look up reserve values ----------------------

	res, err := client.Request(&server.InfoRequest{})
	if err != nil {
		panic(err)
	}
	var serverInfo server.InfoResponse
	if err := res.GetResult(&serverInfo); err != nil {
		panic(err)
	}

	baseReserve := serverInfo.Info.ValidatedLedger.ReserveBaseXRP
	reserveInc := serverInfo.Info.ValidatedLedger.ReserveIncXRP

	fmt.Printf("Base reserve: %v XRP\n", baseReserve)
	fmt.Printf("Incremental reserve: %v XRP\n", reserveInc)

// Look up owner count ----------------------

	address := types.Address("rHb9CJAWyB4rj91VRWn96DkukG4bwdtyTh")
	accountInfo, err := client.GetAccountInfo(&account.InfoRequest{Account: address})
	if err != nil {
		panic(err)
	}

	ownerCount := accountInfo.AccountData.OwnerCount

// Calculate total reserve ----------------------

	totalReserve := float64(baseReserve) + (float64(ownerCount) * float64(reserveInc))

	fmt.Printf("Owner count: %v\n", ownerCount)
	fmt.Printf("Total reserve: %v XRP\n", totalReserve)
}